src/main/java/com/icthh/xm/ms/dashboard/repository/impl/IdRefreshableRepository.java

```diff
@@ -0,0 +1,169 @@
+import java.util.concurrent.atomic.AtomicReference;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.util.AntPathMatcher;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class IdRefreshableRepository implements RefreshableConfiguration, IdRepository {
+
+    private final Map<String, DashboardCounterState> dashboards = new ConcurrentHashMap<>();
+
+    private final AntPathMatcher matcher = new AntPathMatcher();
+    private final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    private final ApplicationProperties applicationProperties;
+    private final TenantContextHolder tenantContextHolder;
+    private final TenantConfigRepository tenantConfigRepository;
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+
+        MsConfigStorageProperties msConfigProperties = applicationProperties.getStorage().getMsConfig();
+        String specificationPathPattern = msConfigProperties.getTenantDashboardPropertiesIdPathPattern();
+        try {
+            String tenant = matcher.extractUriTemplateVariables(specificationPathPattern, updatedKey).get("tenantName");
+            if (isBlank(config)) {
+                dashboards.remove(tenant);
+                log.info("Specification for tenant {} was removed", tenant);
+                return;
+            }
+            String oldHash = DigestUtils.sha1Hex(config);
+            DashboardIdDto dashboardIdDto = mapper.readValue(config, DashboardIdDto.class);
+            DashboardCounterState state = getDashboardCounterState(tenant);
+            state.globalCounterState.set(new DashboardGlobalCounterState(dashboardIdDto.getId(), oldHash));
+            log.info("Specification was for tenant {} updated", tenant);
+        } catch (Exception e) {
+            log.error("Error read specification from path " + updatedKey, e);
+        }
+    }
+
+    @Override
+    public boolean isListeningConfiguration(String updatedKey) {
+        MsConfigStorageProperties msConfigProperties = applicationProperties.getStorage().getMsConfig();
+        String specificationPathPattern = msConfigProperties.getTenantDashboardPropertiesIdPathPattern();
+        return matcher.match(specificationPathPattern, updatedKey);
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        onRefresh(configKey, configValue);
+    }
+
+    @Override
+    @SneakyThrows
+    public Long getNextId() {
+        String tenant = getTenantKeyValue();
+        DashboardCounterState dashboardCounterState = getDashboardCounterState(tenant);
+        DashboardLocalCounter localCounter = dashboardCounterState.getLocalCounter();
+        Optional<Long> counterValue = localCounter.getNextCounter();
+        if (counterValue.isPresent()) {
```

Comment:
why do not use standard Optional.orElseGet()?

Reply:
changed